src/sentry/notifications/notifications/organization_request/base.py

```diff
@@ -65,3 +65,23 @@ def get_log_params(self, recipient: Team | User) -> MutableMapping[str, Any]:
             "user_id": self.requester.id,
             "target_user_id": recipient.id,
         }
+
+    @property
+    def analytics_event(self) -> str | None:
+        """
+        Override this property with the desired name of an analytics event and
+        it will be recorded in `notify()`. Otherwise, only the generic analytics
+        will be recorded by the providers.
+        """
+        return None
+
+    def record_notification_sent(self, recipient: Team | User, provider: ExternalProviders) -> None:
```

Comment:
@mgaeta why are we removing this from the base class?

Reply:
@mgaeta can we avoid modifying `record_notification_sent`? It's already a big PR